app/models/report.rb

```diff
@@ -118,4 +262,111 @@ def table_copyright(collection)
+    }
+  end
+
+  def table_license(collection)
+    result = {}
+    collection.group(:license).count.each do |record|
+      if record[0].instance_of? NilClass
+        result['Undefined'] = record[1]
+      else
+        result[record[0].display_description] = record[1]
+      end
+    end
+    table_populate_defaults result, License.pluck(:display_description)
+    {
+      'title' => 'Thesis counts by Creative Commons license',
+      'summary' => 'This table presents a summary of which Creative Commons license has been selected by the author, '\
+                   'for those theses for which the author retains copyright. The second column gives the specific CC '\
+                   'license selected, while the first column shows how many theses have selected it.',
+      'note' => 'Please note: theses for which the author does not claim copyright will have "Undefined" in this '\
+                'field.',
+      'column' => 'License',
+      'data' => result
+    }
+  end
+
+  def table_hold(collection)
+    result = collection.joins(:hold_sources).group(:source).count
+    table_populate_defaults result, HoldSource.pluck(:source)
+    {
+      'title' => 'Hold counts by hold source',
+      'column' => 'Source',
+      'data' => result
+    }
+  end
+
+  def table_populate_defaults(data, values)
+    values.each do |value|
+      data[value] = '0' unless data[value]
+    end
+  end
+
+  def table_status(collection)
+    result = collection.group(:publication_status).count
+    table_populate_defaults result, Thesis.publication_statuses
+    {
+      'title' => 'Thesis counts by publication status',
+      'summary' => 'This table presents a summary of thesis records by their publication status. The second column '\
+                   'gives the status, while the first column gives how many records have that status.',
+      'column' => 'Publication status',
+      'data' => result
+    }
+  end
+
+  def term_data(collection, term)
+    result = {}
+    result['overall'] = card_overall collection, term
+    result['files'] = card_files collection, term
+    result['issues'] = card_issues collection
+    result['multiple-authors'] = card_multiple_authors collection
+    result['multiple-degrees'] = card_multiple_degrees collection
+    result['multiple-departments'] = card_multiple_departments collection
+    result
+  end
+
+  def term_tables(collection)
+    result = {}
+    result['status'] = table_status collection
+    result['copyright'] = table_copyright collection
+    result['license'] = table_license collection
+    result['department'] = table_department collection
+    result['hold'] = table_hold collection
+    result
+  end
+
+  private
+
+  def pad_category(array)
+    # This builds out the initial set of a category of rows, based on a provided array (like
+    # Thesis.publication_statuses).
+    output = {}
+    array.each do |item|
+      output[item] = {}
+    end
+    output
+  end
+
+  def pad_row(data)
```

Comment:
This feels like it is doing more than padding and is actually populating rows with data? (possibly unlike `pad_category`?)